lib/helpers/deployments.rb

```diff
@@ -1,17 +1,26 @@
+require 'plugins/mcollective'
+
 module Deployments # rubocop:disable Style/Documentation
   def deploy(branch, deleted)
-    if settings.use_mco_ruby
-      result = mco(branch).first
-      raise result.results[:statusmsg] unless result.results[:statuscode].zero?
+    # Currently requires puppet_webhook to be run as a non-root
+    # user with access to running the MCollective client.
+    if settings.use_mcollective
+      results = PuppetWebhook::Mcollective.new('r10k',
+                                               'deploy',
+                                               {
+                                                 dtimeout: settings.discovery_timeout,
+                                                 timeout: settings.client_timeout
+                                               },
+                                               settings.client_cfg,
+                                               environment: branch).run
+      results.each do |result|
+        raise result.results[:statusmsg] unless result.results[:statuscode].zero?
+      end
+      raise results.stats[:noresponsefrom] unless result.stats[:noresponsefrom].length.zero?
```

Comment:
seems good, though raising an array is super weird :P

Reply:
Each RPC Result object is stored as an array and I couldn't think of a better way (at the time) to validate that each result status code was 0.